common/test/acceptance/tests/lms/test_account_settings.py

```diff
@@ -110,15 +123,16 @@ def test_page_view_event(self):
         And I visit my account settings page
         Then a page view analytics event should be recorded
         """
-        self.verify_events_of_type(
-            self.username,
-            u"edx.user.settings.viewed",
-            [{
-                u"user_id": long(self.user_id),
-                u"page": u"account",
-                u"visibility": None,
-            }]
-        )
+
+        event_filter = {
+            'event_type':'edx.user.settings.viewed',
+            'event': {
+                'user_id': self.user_id,
+                'page': 'account',
+                'visibility': None
+            }
+        }
+        self.wait_for_events(event_filter=event_filter, number_of_matches=1)
```

Comment:
Maybe this can be achieved by having a separate argument for event_filter and for event_test.  The former would select the events you wish to consider, but the second is actually applied as part of the assert.  Instead of "wait_for_events", something like "assert_matching_events_occur"?  In the above example, only the event_type would be used for filtering, and the 'event' clause could be used for testing.  (Or call it event_assert instead of event_test.)